storage/bootstrap/result/result.go

```diff
@@ -65,8 +65,22 @@ func (r *bootstrapResult) AddResult(other BootstrapResult) {
 	if other == nil {
 		return
 	}
-	r.results.AddResults(other.ShardResults())
-	r.unfulfilled.AddRanges(other.Unfulfilled())
+	size := 0
+	sizeOther := 0
+	for _, sr := range r.results {
+		size += len(sr.AllSeries())
+	}
+	for _, sr := range other.ShardResults() {
+		sizeOther += len(sr.AllSeries())
+	}
+	if size >= sizeOther {
+		r.results.AddResults(other.ShardResults())
+		r.unfulfilled.AddRanges(other.Unfulfilled())
+	} else {
+		other.AddResult(r)
```

Comment:
Can change the naming/or something to reflect this perhaps?  This was a definite issue we saw though when merging the zero sized FS bootstrap result with a huge Peers bootstrap result.  We basically added each series one by one into the empty FS bootstrap result.  

Definitely want to avoid this case.